packages/path_provider/path_provider/lib/path_provider.dart

```diff
@@ -5,21 +5,12 @@
 import 'dart:async';
 import 'dart:io' show Directory;
 
-import 'package:flutter/services.dart';
-import 'package:meta/meta.dart';
-import 'package:platform/platform.dart';
+import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
 
-const MethodChannel _channel =
-    MethodChannel('plugins.flutter.io/path_provider');
+export 'package:path_provider_platform_interface/path_provider_platform_interface.dart'
+    show StorageDirectory;
 
-Platform _platform = const LocalPlatform();
-
-/// This API is only exposed for the unit tests. It should not be used by
-/// any code outside of the plugin itself.
-@visibleForTesting
-void setMockPathProviderPlatform(Platform platform) {
-  _platform = platform;
-}
```

Comment:
This may be a backwards incompatible change for the plugin. In the `connectivity` plugin, apparently, some tests in our internal repo broke (probably because they were overriding, or calling methods in the internal MethodChannel instance.

I'd leave this method in the final version, just in case.